### PR TITLE
Add docs about cairo 1 modules

### LINKED ISSUE #1591: Consider documenting making cairo1 modules
We should definetely document how to add a cairo module to our `protostar.toml`, and people should know the concept of modules (how to create/modify them). I would hold this off until some cairo1 documentation is available and we can provide a solid info on that.

### PR DESCRIPTION
**Pull request by Arcticae**
Closes #1591

### CODE REVIEW

File: website/docs/tutorials/08-cairo-1-support/01-project-initialization.md [hunk 1]
```diff
@@ -34,6 +34,59 @@ This template contains:
     - Single test file with one test for function defined in `main.cairo`
 - `protostar.toml` containing information about the project
 
+## Cairo 1 modules
+
+In order to understand how to create cairo1 modules, we need to talk about `cairo_project.toml` and `lib.cairo` purpose.
```
Comment (mkaput) [inline website/docs/tutorials/08-cairo-1-support/01-project-initialization.md:39]:
```suggestion
In order to understand how to create Cairo 1.0 modules, we need to talk about `cairo_project.toml` and `lib.cairo` purpose.
```

File: website/docs/tutorials/08-cairo-1-support/01-project-initialization.md [hunk 2]
```diff
@@ -34,6 +34,59 @@ This template contains:
...
+### Project defaults
+#### 1. `cairo_project.toml`
```
Comment (mkaput) [inline website/docs/tutorials/08-cairo-1-support/01-project-initialization.md:41]: Shouldn't cairo_project be autogenerated by Protostar? That's what's happening in Scarb

**Review by mkaput — CHANGES_REQUESTED** [review]

File: website/docs/tutorials/08-cairo-1-support/01-project-initialization.md (hunk 2 above)
Comment (Arcticae) [reply website/docs/tutorials/08-cairo-1-support/01-project-initialization.md:41]: Yes it is - that's why it's listed under "project defaults"

Comment (Arcticae) [reply website/docs/tutorials/08-cairo-1-support/01-project-initialization.md:41]: I mean, we don't add to this file automatically if you meant that

**Review by mkaput — CHANGES_REQUESTED** [review]

File: website/docs/tutorials/08-cairo-1-support/01-project-initialization.md [hunk 3]
```diff
@@ -34,6 +34,59 @@ This template contains:
     - Single test file with one test for function defined in `main.cairo`
 - `protostar.toml` containing information about the project
 
+## Cairo 1 modules
+
+In order to understand how to create Cairo 1.0 modules, we need to talk about `cairo_project.toml` and `lib.cairo` purpose.
```
Comment (karol-bisztyga) [inline website/docs/tutorials/08-cairo-1-support/01-project-initialization.md:43]:
```suggestion
In order to understand how to create Cairo 1.0 modules, we need to talk about the purpose of `cairo_project.toml` and `lib.cairo`.
```

File: website/docs/tutorials/08-cairo-1-support/01-project-initialization.md [hunk 4]
```diff
@@ -34,6 +34,59 @@ This template contains:
...
+### Project defaults
+#### 1. `cairo_project.toml`
+It is needed for definition of crate roots, so the places where `lib.cairo` files are located
```
Comment (karol-bisztyga) [inline website/docs/tutorials/08-cairo-1-support/01-project-initialization.md:46]:
```suggestion
It is needed for the definition of crate roots, which are the places where `lib.cairo` files are located.
```

File: website/docs/tutorials/08-cairo-1-support/01-project-initialization.md [hunk 5]
```diff
@@ -34,6 +34,59 @@ This template contains:
...
+
+Content of the default `cairo_project.toml` contains only definition of `src` crate
```
Comment (karol-bisztyga) [inline website/docs/tutorials/08-cairo-1-support/01-project-initialization.md:48]:
```suggestion
The default `cairo_project.toml` file contains only the definition of the `src` crate 
```

File: website/docs/tutorials/08-cairo-1-support/01-project-initialization.md [hunk 6]
```diff
@@ -34,6 +34,59 @@ This template contains:
...
+#### 1. `cairo_project.toml`
+It is needed for definition of crate roots, so the places where `lib.cairo` files are located
+
+Content of the default `cairo_project.toml` contains only definition of `src` crate 
+```toml
+[crate_roots]
+src = "."
+```
+
+The `src` crate is then imported in our tests in following manner:
```
Comment (karol-bisztyga) [inline website/docs/tutorials/08-cairo-1-support/01-project-initialization.md:54]:
```suggestion
The `src` crate is then imported into our tests in the following manner:
```

File: website/docs/tutorials/08-cairo-1-support/01-project-initialization.md [hunk 7]
```diff
@@ -34,6 +34,59 @@ This template contains:
...
+```
+use src::main::fib;
+```
+
+#### 2. `lib.cairo`
+It is the root of the module tree of the package. here you can define functions, declare used modules, etc.
+
+The default one has only `main` module declaration:
```
Comment (karol-bisztyga) [inline website/docs/tutorials/08-cairo-1-support/01-project-initialization.md:66]:
```suggestion
The default one has only the `main` module declaration:
```

File: website/docs/tutorials/08-cairo-1-support/01-project-initialization.md [hunk 8]
```diff
@@ -34,6 +34,59 @@ This template contains:
...
+```
+mod main;
+```
+### Creating and using a new module
+
+Suppose we wanted to create a module called `mod1` inside the `src` crate, and use it in tests.
```
Comment (karol-bisztyga) [inline website/docs/tutorials/08-cairo-1-support/01-project-initialization.md:72]:
```suggestion
Suppose we wanted to create a module called `mod1` inside the `src` crate and use it in tests.
```

File: website/docs/tutorials/08-cairo-1-support/01-project-initialization.md [hunk 9]
```diff
@@ -34,6 +34,59 @@ This template contains:
...
+src = "."
+```
+
+The `src` crate is then imported in our tests in following manner:
+```
+use src::main::fib;
+```
+
+#### 2. `lib.cairo`
+It is the root of the module tree of the package. here you can define functions, declare used modules, etc.
```
Comment (karol-bisztyga) [inline website/docs/tutorials/08-cairo-1-support/01-project-initialization.md:64]:
```suggestion
It is the root of the module tree of the package. Here you can define functions, declare used modules, etc.
```

File: website/docs/tutorials/08-cairo-1-support/01-project-initialization.md [hunk 10]
```diff
@@ -34,6 +34,59 @@ This template contains:
...
+### Creating and using a new module
+
+Suppose we wanted to create a module called `mod1` inside the `src` crate, and use it in tests.
+
+Here are the steps we need to do:
+
+1. Create `mod1` directory in `src` crate
+2. Create `mod1.cairo` alongside this directory. 
+3. Create your source file inside of `mod1` (i.e. `functions.cairo` or any suitable name). Define your code here.
+4. Declare the source file/files in `mod1.cairo`. The file contents should look like this (assuming you have `functions.cairo` from previous step):
```
Comment (karol-bisztyga) [inline website/docs/tutorials/08-cairo-1-support/01-project-initialization.md:79]:
```suggestion
Here are the steps we need to take:

1. Create a `mod1` directory in the `src` crate
2. Create `mod1.cairo` alongside this directory. 
3. Create your source file inside of `mod1` (i.e. `functions.cairo` or any suitable name). Define your code here.
4. Declare the source file/files in `mod1.cairo`. The file contents should look like this (assuming you have `functions.cairo` from the previous step):
```

**Review by mkaput — APPROVED** [review]

File: website/docs/tutorials/08-cairo-1-support/01-project-initialization.md (hunk 6 above)
Comment (Arcticae) [reply website/docs/tutorials/08-cairo-1-support/01-project-initialization.md:54]: That sounds incorrect to me

**Pull request merged**